docs/regression_test_api.rst

```diff
@@ -86,42 +111,86 @@ In essence, these builtins exert control over the test creation, and they allow
      This only has an effect if used with ``inherit_params=True``.
 
 
-.. py:function:: reframe.core.pipeline.RegressionTest.variable(*types, value=None, field=None)
+.. py:function:: reframe.core.pipeline.RegressionTest.variable(*types, value=None)
 
   Inserts a new regression test variable.
+  Declaring a test variable through the :func:`variable` built-in allows for a more robust test implementation than if the variables were just defined as regular test attributes (e.g. ``self.a = 10``).
+  Using variables declared through the :func:`variable` built-in guarantees that these regression test variables will not be redeclared by any child class, while also ensuring that any values that may be assigned to such variables comply with its original declaration. 
+  In essence, by using test variables, the user removes any potential test errors that might be caused by accidentally overriding a class attribute. See the example below.
+
 
   .. code:: python
 
     class Foo(rfm.RegressionTest):
-        my_var = variable(int, value = 8)
+        my_var = variable(int, value=8)
+        not_a_var = 4
 
         def __init__(self):
             print(self.my_var) # prints 8.
+            # self.my_var = 'override' # Error: my_var must be an int!
+            self.not_a_var = 'override' # However, this would work. Dangerous!
 
-  The argument ``value`` sets the default value for the variable.
-  A variable may not be declared more than once. However, it is possible to alter a variable's value after it was declared by simply assigning it a new value directly in the class body.
+  The argument ``value`` in the :func:`variable` built-in sets the default value for the variable.
+  As mentioned above, a variable may not be declared more than once, but its value can be updated by simply assigning it a new value directly in the class body.
```

Comment:
```suggestion
  As mentioned above, a variable may not be declared more than once, but its default value can be updated by simply assigning it a new value directly in the class body.
```

Users might think that this is the only way to update a variable, whereas they can do that in their tests as well. I think we should show that, too. Perhaps by simply adding another assignment to the example above:

```python
self.my_var = 10 # tests may also assign values the standard way
```

Reply:
I see that you mention it later, but it wouldn't hurt here, too. I leave it up to you.